### ISSUE TITLE
Callstack Instrument fails to capture some parts of the stack

### ISSUE THREAD

**Issue by vringar** [mozilla/OpenWPM#845]
@hadiamjad pointed out on Matrix that the callstack instrument captures less than the Firefox debugger shows.

![Firefox output](https://mozilla.modular.im/_matrix/media/r0/download/matrix.org/SPWjnxCWVCpPKVHBEttJYesK)
![OpenWPM output](https://mozilla.modular.im/_matrix/media/r0/download/matrix.org/oBAIhNfiJDxhKXXVKCEtIgJc]

**Comment by vringar** [comment]
It would be nice to get the raw string output from OpenWPM instead of an image since that's easier to read/understand (at least for me).

**Comment by hadiamjad** [comment]
2/a</e.prototype.addCreative@https://ced-ns.sascdn.com/diff/templates/ts/dist/banner/sas-banner-1.2.js:1:16788;null
16/o</e.prototype.init@https://ced-ns.sascdn.com/diff/templates/ts/dist/banner/sas-banner-1.2.js:1:13108;null
onLoad@https://www3.smartadserver.com/ac?nwid=0&siteid=81298&pgid=632202&fmtid=35212&async=1&visit=s&tmstp=6304051897&tag=sas_35212&sh=768&sw=1366&pgDomain=https%3A%2F%2Fwww.atresplayer.com%2F&noadcbk=sas.noad&isLazy=0&isAdRefresh=0:111:27;null
l@https://www3.smartadserver.com/ac?nwid=0&siteid=81298&pgid=632202&fmtid=35212&async=1&visit=s&tmstp=6304051897&tag=sas_35212&sh=768&sw=1366&pgDomain=https%3A%2F%2Fwww.atresplayer.com%2F&noadcbk=sas.noad&isLazy=0&isAdRefresh=0:58:1483;null
sas.utils.Latch/this.notify@https://www3.smartadserver.com/ac?nwid=0&siteid=81298&pgid=632202&fmtid=35212&async=1&visit=s&tmstp=6304051897&tag=sas_35212&sh=768&sw=1366&pgDomain=https%3A%2F%2Fwww.atresplayer.com%2F&noadcbk=sas.noad&isLazy=0&isAdRefresh=0:58:1639;null
sas.utils.loadScript/r</<@https://www3.smartadserver.com/ac?nwid=0&siteid=81298&pgid=632202&fmtid=35212&async=1&visit=s&tmstp=6304051897&tag=sas_35212&sh=768&sw=1366&pgDomain=https%3A%2F%2Fwww.atresplayer.com%2F&noadcbk=sas.noad&isLazy=0&isAdRefresh=0:58:3392;null
e/sas.utils._callbacks[c]<@https://www3.smartadserver.com/ac?nwid=0&siteid=81298&pgid=632202&fmtid=40121&async=1&visit=s&tmstp=6304051897&tag=sas_40121&sh=768&sw=1366&pgDomain=https%3A%2F%2Fwww.atresplayer.com%2F&noadcbk=sas.noad&isLazy=0&isAdRefresh=0:58:2866;null
sas.events.addLoadEvent/e<@https://www3.smartadserver.com/ac?nwid=0&siteid=81298&pgid=632202&fmtid=40121&async=1&visit=s&tmstp=6304051897&tag=sas_40121&sh=768&sw=1366&pgDomain=https%3A%2F%2Fwww.atresplayer.com%2F&noadcbk=sas.noad&isLazy=0&isAdRefresh=0:58:1260;null
r@https://statics.atresmedia.com/atresplayer/webapp/vendor.873e38b533.js:9:76012;null

**Comment by hadiamjad** [comment]
this might be the reason: https://github.com/mozilla/OpenWPM/issues/678?

**Comment by vringar** [comment]
Yes, that seems like the reason. Can you check if that setting is set?

**Comment by hadiamjad** [comment]
Where I can find this setting?

**Comment by vringar** [comment]
https://github.com/mozilla/OpenWPM/blob/8dad9fd2129f8375575813baaa19992efdde47de/test/test_profile.py#L109-L134
If you add this command to the command sequence you can check whether the pref is set.

**Comment by vringar** [comment]
I just realized that this answer is a little short. Should I explain myself further or is this enough to get you going?

**Comment by hadiamjad** [comment]
No, it's understandable, I'll update you on this soon.

**Comment by hadiamjad** [comment]
Here, I have added TestConfigSetCommand at line 68 and it raises AssertionError. Looks like flag is not set.

![Screenshot (102)](https://user-images.githubusercontent.com/46374292/105571731-3dd6aa00-5d74-11eb-8182-93036e1e97f8.png)

**Comment by vringar** [comment]
You can set the pref by using the BrowserParams like this: 
```python
browser_params[i].prefs = {"javascript.options.asyncstack": True}
```

**Comment by hadiamjad** [comment]
Nope, unfortunately, it's still not working.

**Comment by hadiamjad** [comment]
Interestingly when I tried capturing the call stack directly from selenium, it has the same problem. 
![image](https://user-images.githubusercontent.com/46374292/106161563-fc496300-61a8-11eb-936f-8f450014172a.png)

**Comment by vringar** [comment]
Hmm, could you file a bug with Selenium then to see if maybe they can fix this?
